model/DataQuery.php

```diff
@@ -702,6 +705,15 @@ public function applyRelation($relation) {
 				$this->query->addLeftJoin($componentBaseClass,
 					"\"$relationTable\".\"$componentField\" = \"$componentBaseClass\".\"ID\"");
 				}
+				else {
+					$alias = uniqid($componentBaseClass);
+					$this->query->addLeftJoin(
```

Comment:
So we just alias as an unintelligible string? Doesn't seem very fluent to me... you should see my feedback on https://github.com/silverstripe/silverstripe-framework/pull/5531/files#r63108288